File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        seen = set(nums)
        mxcnt = 0

        for num in nums:
            seen.add(num)
        
        for num in nums:
            if num - 1 not in seen:
                curr = num
                cnt = 1

                while curr +1 in seen:
                    cnt += 1
                    curr += 1
                
                mxcnt = max(mxcnt, cnt)
        
        return mxcnt

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-1.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int n = nums.size();
        int cnt = 0;
        for (int i=0;i<n;i++){
            int j = i;
            int currsum = 0;
            while (j<n){
                currsum += nums[j];
                if (currsum == k){
                    cnt++;
                }
                j++;
            }
        }
        return cnt;
    }
};